src/dali_backend.cc

```diff
@@ -220,6 +220,10 @@ class DaliModelInstance : public ::triton::backend::BackendModelInstance {
     return ret;
   }
 
+  int32_t DetermineDeviceId() {
+    return !CudaStream() ? -1 : device_id_;
```

Comment:
```suggestion
    return !CudaStream() ? ::dali::CPU_ONLY_DEVICE_ID : device_id_;
```

Reply:
I'd prefer to retain the `-1`. I believe this code shouldn't really depend on DALI, cause it's heavily dependent on Triton. I'd prefer having it separated, just in case any changes are needed

Reply:
Btu this device ID is for DALI not for the rest TRITON.

Reply:
I'd prefer keeping `dali_backend.cc` separated from DALI's API. We do use `DeviceGuard` here, but that can be easily replaced. And I'd prefer to have all DALI API details encaplusated inside `DaliPipeline` class. I'll discuss it with @banasraf

Reply:
Well, if the backend used DALI then you cannot separate it. I assume you may need to use DetermineDeviceId valu inside DALI one day so I would prefer to use proper values.

Reply:
I don't see any reason to limit our use of DALI's API. We use multiple parts of it across all of the backend (e.g. TensorListShape is used everywhere). Using DALI stuff just makes coding so much easier (DeviceBuffer, DeviceGuard, CUDAStream, UniqueHandle....) and I don't see any value in treating DALI backend as something separate from DALI.

Reply:
Sounds fair, I'll apply the change